standard/interfaces.md

```diff
@@ -230,92 +230,92 @@ interface_member_declaration
     ;
 ```
 
-An interface declaration declares zero or more members. The members of an interface shall be constants, fields, methods, properties, events, indexers, operators, constructors, and types, some of which may be instance, others static, as described in the subclauses for each interface member kind. Interfaces shall not declare instance data members, including field like events.
+This clause augments the description of members in classes (§15.3) with restrictions for interfaces. The Interface members are declared using *member_declaration*s with the following additional rules:
 
-All interface members implicitly have public access; however, an explicit access modifier ([§7.5.2](basic-concepts.md#752-declared-accessibility)) is permitted except on static constructors (§15.12).
+- A *finalizer_declaration* is not allowed.
+- Instance constructors, *constructor_declaration*s, are not allowed.
+- All interface members implicitly have public access; however, an explicit access modifier ([§7.5.2](basic-concepts.md#752-declared-accessibility)) is permitted except on static constructors (§15.12).
+- An interface instance function member whose declaration includes a body is an implicitly `virtual` member unless the `sealed` or `private` modifier is used.
+- The `virtual` modifier may be used on a function member that would otherwise be implicitly `virtual`.
+- The `abstract` modifier is implied for interface function members without bodies; that modifier may be given explicitly.
+- A derived interface may override an abstract member declared in a base interface. A non-virtual instance function member may be declared using the `sealed` keyword.
```

Comment:
(I'm used to using the term 'virtual' in the sense of 'virtual call.' If that is acceptable here, then the term 'virtual' covers abstract members as well. If not, then I guess we should mention both?)
```suggestion
- A derived interface may override a virtual member declared in a base interface. A non-virtual instance function member may be declared using the `sealed` keyword.
```
Also, I believe the only way to override is to provide an explicit implementation; otherwise, you end up declaring a new virtual member (and the `override` keyword is not supported). Should the means of overriding be clarified here?

Reply:
Notes for a commit tomorrow to address all @jnm2 's comments regarding *abstract* vs. *non-virtual*, and the comments regarding the accessibility of the name.

1. An explicit interface implementation is the only way to provide an override of a member defined in an interface. Any other declaration (`virtual` or otherwise) is a new member. It doesn't *hide* the member defined in an interface because of the accessibility rules...
2. An instance interface member is `virtual` (or `abstract`) by default. To prevent implementing types or derived interfaces from providing an overriding implementation, it must be declared `sealed`.
3. A member defined in an interface is accessible only through a reference to that interface type. I'm not sure "visible" is the right term.